components/brave_shields/resources/panel/components/main-panel/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react'
+import getPanelBrowserAPI, { UIHandlerReceiver } from '../../api/panel_browser_api'
+
+export function useGetShieldsData() {
+  const [url, setUrl] = React.useState("")
+  const [totalBlockedResources, setTotalBlockedResources] = React.useState(0)
```

Comment:
Is there a benefit to split up each piece of siteData? How about just a single `React.useState<SiteInfo>` since we get a new piece of data each time?